MCprep_addon/materials/generate.py

```diff
@@ -1147,182 +1354,183 @@ def matgen_cycles_principled(mat, passes, use_reflections, only_solid):
+		nodeGlossDiff.inputs["Roughness"].default_value = 0.7
+	
+	# Sets default metallic values
+	if use_reflections and checklist(canon, "metallic"):
+		nodeMixMetallic.inputs["Fac"].default_value = 1
+		
+		if nodeGlossMetallic.inputs["Roughness"].default_value < 0.2:
+			nodeGlossMetallic.inputs["Roughness"].default_value = 0.2
+		if nodeMathPower.inputs[0].default_value < 0.2:
+			nodeMathPower.inputs[0].default_value = 0.2
+		if nodeGlossDiff.inputs["Roughness"].default_value < 0.2:
+			nodeGlossDiff.inputs["Roughness"].default_value = 0.2
+	
+	else:
+		nodeMixMetallic.inputs["Fac"].default_value = 0
+
+	# Connect nodes
+	links.new(nodeMixDiff.outputs["Shader"], nodeMixMetallic.inputs[1])
+	links.new(nodeMathPower.outputs[0], nodeMixRGB.inputs["Fac"])
+	links.new(nodeMathPower.outputs[0], nodeDiff.inputs["Roughness"])
+	links.new(nodeMixRGB.outputs[0], nodeFresnel.inputs["Normal"])
+	links.new(nodeFresnel.outputs[0], nodeMixRGBDiff.inputs["Fac"])
+	links.new(nodeGeometry.outputs["Incoming"], nodeMixRGB.inputs["Color2"])
+	links.new(nodeBump.outputs["Normal"], nodeMixRGB.inputs["Color1"])
+	links.new(nodeMathPowerDiff.outputs["Value"], nodeMixRGBDiff.inputs["Color1"])
+	links.new(nodeMixRGBDiff.outputs["Color"], nodeMathMultiplyDiff.inputs[0])
+	links.new(nodeMathMultiplyDiff.outputs["Value"], nodeMixDiff.inputs["Fac"])
+	links.new(nodeDiff.outputs["BSDF"], nodeMixDiff.inputs[1])
+	links.new(nodeGlossDiff.outputs["BSDF"], nodeMixDiff.inputs[2])
+	links.new(nodeFresnelMetallic.outputs["Fac"], nodeMixRGBMetallic.inputs["Fac"])
+	links.new(nodeMathMetallic.outputs["Value"], nodeMixRGBMetallic.inputs["Color2"])
+	links.new(nodeMixRGBMetallic.outputs["Color"], nodeGlossMetallic.inputs["Color"])
+	links.new(nodeGlossMetallic.outputs["BSDF"], nodeMixMetallic.inputs[2])
+	links.new(nodeMixMetallic.outputs["Shader"], nodeMixEmit.inputs[1])
+	links.new(nodeMixCam.outputs["Shader"], nodeMixEmit.inputs[2])
+	links.new(nodeTrans.outputs["BSDF"], nodeMixTrans.inputs[1])
+	links.new(nodeFalloff.outputs["Linear"], nodeEmit.inputs["Strength"])
+	links.new(nodeLightPath.outputs["Is Camera Ray"], nodeMixCam.inputs["Fac"])
+	links.new(nodeEmit.outputs["Emission"], nodeMixCam.inputs[1])
+	links.new(nodeEmitCam.outputs["Emission"], nodeMixCam.inputs[2])
+	links.new(nodeMixEmit.outputs["Shader"], nodeMixTrans.inputs[2])
+	links.new(nodeMixTrans.outputs["Shader"], nodeOut.inputs["Surface"])
+	
+	nodeInputs = [[nodeMixRGBMetallic.inputs["Color1"], nodeMathMetallic.inputs[0], nodeDiff.inputs["Color"], nodeEmit.inputs["Color"], nodeEmitCam.inputs["Color"]], [nodeMixTrans.inputs["Fac"]], [nodeMixEmit.inputs[0]], [nodeGlossDiff.inputs["Roughness"], nodeGlossMetallic.inputs["Roughness"], nodeMathPower.inputs[0]], [nodeMixMetallic.inputs["Fac"]], [nodeMathPowerDiff.inputs[0]], [nodeDiff.inputs["Normal"], nodeGlossMetallic.inputs["Normal"], nodeFresnelMetallic.inputs["Normal"], nodeGlossDiff.inputs["Normal"], nodeBump.inputs["Normal"]]]
```

Comment:
Completely agree with you! For now as a quick fix, I've just split it into separate lines since I am busy this week. Additionally, it would probably be faster to leave this for now and fix it in a separate commit later as the current should be functional now.

Reply:
Great, thanks so much - and agreed, functional now is the key thing!